spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceBucket.java

```diff
@@ -52,16 +53,26 @@ public GoogleStorageResourceBucket(Storage storage, String bucketName,
 
 		this.storage = storage;
 		this.bucketName = bucketName;
-		this.createBucketIfRequired = createBucketIfNotExists;
-	}
+		this.createBucketIfNotExists = createBucketIfNotExists;
+	}
+
+	/**
+	 * Creates the bucket that this {@link GoogleStorageResourceBucket} represents in Google Cloud
+	 * Storage.
+	 * @return the {@link com.google.cloud.storage.Blob} of the created object or {@code null} if
+	 * the bucket already exists or cannot be created
+	 */
+	public Bucket create() {
+		if (exists() || !this.createBucketIfNotExists) {
+			return null;
```

Comment:
I'm not a fan of returning `null` here.
One alternative is to throw an `IllegalStateException`.
Another alternative is to silently return the existing bucket, and clarify in the javadoc that the method only creates the bucket if it doesn't exist.
I'm leaning more towards the latter.

Reply:
I don't think they're mutually exclusive. I actually think they're complementary.
If bucket already exists, return it, if not and it can't be created, do what? Return `null` or throw exception?
I'm more inclined towards returning `null` since the reason the bucket can't be created is within the user control (e.g., change `createBucketIfNotExists` to `true` and it will start working).

Reply:
OK, I kind of agree with you that maybe 'getOrCreate()` is not the behavior this method implies.
However, I think that returning `null` is a poor way of communicating a user error. Throwing `IllegalStateException` is more direct and easier to debug for the user.

Reply:
I think what I was trying to say is that I don't consider this case to be an "error" -- rather a choice by a user who specified auto-create as `false`.
Because it's not an error, but rather the desired behaviour, I don't think an exception should be raised.
In other places, e.g., if we're trying to write to a file and it doesn't exist, an exception should possibly be raised.

Reply:
I was focusing on the first part of the condition `exists()`. In fact, this method should create the bucket whether `createBucketIfNotExists` is true or not. It shouldn't matter because it's the user's explicit intent to create the bucket.